Unity/PetEver/Assets/02.Scripts/TreeController.cs

```diff
@@ -53,28 +61,84 @@ private void Awake()
 
         void Start()
         {
-            treePrefab = Resources.Load<GameObject>("Prefabs/tree5");
+            LoadTreeResoures();
+            initTreeButtons();
             treePopupCanvasGroup = GameObject.Find("TreePopupPannel").GetComponent<CanvasGroup>();
             treeCreatePopupPanelCanvasGroup = GameObject.Find("TreeCreatePopupPannel").GetComponent<CanvasGroup>();
             manCharacter = GameObject.FindGameObjectWithTag("Owner");
+            mainCanvas = GameObject.FindGameObjectWithTag("UICanvas");
             treeZone = GameObject.Find("TreeZone");
             exitBtn = GameObject.Find("ExitBtn");
             exitBtn.GetComponent<Button>().onClick.AddListener(() =>
             {
-                ExitPopup(treePopupCanvasGroup);
+                ExitOpenedPopup();
             });
             popupBack = GameObject.Find("TreePopupPannel");
             popupBack.GetComponent<Button>().onClick.AddListener(() =>
             {
-                ExitPopup(treePopupCanvasGroup);
+                ExitOpenedPopup();
             });
-            mainCanvas = GameObject.FindGameObjectWithTag("UICanvas");
             createTreeBtn = GameObject.Find("Plus");
             createTreeBtn.GetComponent<Button>().onClick.AddListener(() =>
             {
-                createTree();
+                ShowPopUpDetail(treeCreatePopupPanelCanvasGroup);
+            });
+            selectTreeBtn = GameObject.Find("SelectTreeBtn");
+            selectTreeBtn.GetComponent<Button>().onClick.AddListener(() =>
+            {
+
+                ExitOpenedPopup();
+                if (selectedTree != null)
+                {
+                    createTree(selectedTree);
+                    selectedTree = null;
+                }
             });
+        }
 
+        void initTreeButtons()
+        {
+            int treeBtnSize = 3;
```

Comment:
If this `treeBtnSize` is common for another function, it is better to declare it as global/static. Then it is much better to modify if want to change the size.